Classes/Projection/AbstractBaseProjector.php

```diff
@@ -0,0 +1,87 @@
+<?php
+namespace Neos\Cqrs\Projection;
+
+/*
+ * This file is part of the Neos.Cqrs package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Cqrs\Event\EventInterface;
+use TYPO3\Flow\Log\SystemLoggerInterface;
+use TYPO3\Flow\Reflection\ObjectAccess;
+use TYPO3\Flow\Annotations as Flow;
+
+/**
+ * A base class for projectors
+ *
+ * Specialized projectors may extend this class in order to use the convenience methods included. Alternatively, they
+ * can as well just implement the ProjectorInterface and refrain from extending this base class.
+ *
+ * @api
+ */
+abstract class AbstractBaseProjector implements ProjectorInterface
+{
+
+    /**
+     * @Flow\Inject
+     * @var SystemLoggerInterface
+     * @api
+     */
+    protected $systemLogger;
+
+    /**
+     * Concrete projectors may override this property for setting the class name of the Read Model to a non-conventional name
+     *
+     * @var string
+     * @api
+     */
+    protected $readModelClassName;
+
+    /**
+     * Sets the properties in a Read Model with corresponding properties of an event according to the given map of
+     * event and read model property names.
+     *
+     * If no property names are specified, this method will try to map all accessible properties of the event to
+     * the same name properties in the read model:
+     *
+     * $this->mapEventToReadModel($someEvent, $someModel);
+     *
+     * If you want more control over the mapping, you can pass the property names in three different ways:
+     *
+     * 1. [ "eventPropertyName" => "readModelPropertyName", ... ]
+     * 2. [ "propertyName", ...]
+     *
+     * In the second case "propertyName" will be used both for determining the event property as well as the read model property.
+     * Combinations of both are also possible:
+     *
+     * 3. [ "eventSomeFoo" => "readModelSomeFoo", "bar", "baz", "eventSomeQuux" => "readModelSomeQuux" ]
+     *
+     * For use in the concrete projection.
+     *
+     * @param EventInterface $event An event
+     * @param object $readModel A read model
+     * @param array $propertyNamesMap Property names of the event (key) and of the read model (value). Alternatively just the property name as value.
+     * @return void
+     * @api
+     */
+    protected function mapEventToReadModel(EventInterface $event, $readModel, array $propertyNamesMap = [])
```

Comment:
I don't understand why it should be an issue to have that _optional_ convenience method in the framework. Especially because it can be _combined_ with manual assignment.

I do see a risk of people just blindly making event payload and read model synchronous. But I don't think that that will be because of a convenience method like this. And after all the names within event and projections are part of the UL so it's pretty natural that they use the same vocabulary I guess.

So, while it's true that you won't need that method for some projections I think that for most Aggregates there will be at least one ReadModel that would benefit from that.

I was skeptical too at first, but having to deal with 10+ fields in an event payload I really embrace framework support like this   (I know that we should try to limit the fields but in some scenarios (think of user signup for example) this is just not possible)